src/shared/telemetry/defaultTelemetryService.ts

```diff
@@ -221,8 +219,8 @@ export class DefaultTelemetryService implements TelemetryService {
 
     private injectAccountMetadata(event: TelemetryEvent, awsContext: AwsContext): TelemetryEvent {
         let accountValue: string | AccountStatus
-        if (event.namespace === TelemetryNamespace.Session) {
-            // this matches JetBrains' functionality: the AWS account ID is not set on session start.
+        // The AWS account ID is not set on session start. This matches JetBrains' functionality.
+        if (event.data.every(item => item.name.startsWith('session'))) {
```

Comment:
Is it worth checking "starts with `session.`" in case we did something weird like sessions and didn't want that to be considered the same "namespace"?

Reply:
`session_start` and `session_end` have underscores. I'm kind of tempted to check only `session_start` and `session_end` explicitly, thoughts?

Reply:
That sounds reasonable to me. That makes the code more self explanatory as well (compared to checking against Session as it has been)

Reply:
Sounds good, changed to check specifically `session_start` and `session_end`